R/tostring.R

```diff
@@ -48,13 +48,17 @@ default_hsep <- d_hsep_factory()
 #' of the table-like object reprseented by `x`
 setMethod("toString", "MatrixPrintForm", function(x,
                                                   widths = NULL,
+                                                  indent_titles = FALSE,
```

Comment:
It makes perfect sense. I will add a `wrap` logical and a `max_width` numeric. What do you think should be the default for the second one? and what should be done in case some parts of the text have "words" that are longer than the allocated horizontal space? In my implementation, I was throwing an error and the two things above are related because also my test cases produce only a small table, and if you use `ncchar` as default, the horizontal space allowed is almost always smaller than the added title. I think the best would be to use 80 (terminal std) as default. This will change a lot of tests downstream though (maybe not too many as it is only title and footer)

Reply:
The default should either be nothing, or the current value of `options("width")` in interactive print to the console settings. In any pagination aware context it should instead be the value of `cpp`, if that is set.

You're right that this will break a lot of tests. This doesnt mean we shouldnt do it, but it does mean we need to be careful of when we put this change in. Its probably far enough from a release sprint that we can get away with it, but if not, it would need to wait and be added to to the next (in person) PI planning board. We can develop the feature in a branch in the meantime though so it doesn't necessarily mean we can't work on this.

Also, we can do this in two steps. Put the machinery in but leave the default as is (ie nothing), and then later when we're ready to flip the switch, we simply change the default.

Reply:
As for "words" that take up more than the allotted space, that is indeed an issue we'll need to think about, as `base:strwrap` unfortunately doesn't help us there.  The case I forsee where that may happen is file paths in the provenance footer (it should be exceedingly rare or even impossible with real words in any realistic setting):

```
> strwrap("/my/path/to/something_really/long/that/is/too/long", width = 10)
[1] "/my/path/to/something_really/long/that/is/too/long"
```

Reply:
I think we could start dealing with the above cases (that, as you said, certainly will appear) as soon as they do appear. For the moment, I think we can keep it as it is.